libs/ui-lib/src/lib/custom-text-input/custom-text-input.tsx

```diff
@@ -31,21 +32,36 @@ export const CustomTextInput = ({
   hasErrors,
   errorMessage,
   svgIcon,
+  isPassword,
 }: CustomTextInputProps) => {
   return (
     <div className={`custom-text-input-container ${hasErrors ? 'error' : ''}`}>
       <label className="custom-text-input-label" htmlFor={fieldName}>
         {title}
       </label>
-      <input
-        className={`custom-text-input input-border ${hasErrors ? 'error' : ''}`}
-        id={fieldName}
-        name={fieldName}
-        data-testid="custom-text-input"
-        type="text"
-        value={value}
-        onChange={onChange}
-      />
+
+      {isPassword ? (
+        <input
+          className={`custom-text-input input-border ${hasErrors ? 'error' : ''}`}
+          id={fieldName}
+          name={fieldName}
+          data-testid="custom-text-input"
+          type="password"
+          value={value}
+          onChange={onChange}
+        />
+      ) : (
+        <input
+          className={`custom-text-input input-border ${hasErrors ? 'error' : ''}`}
+          id={fieldName}
+          name={fieldName}
+          data-testid="custom-text-input"
+          type="text"
+          value={value}
+          onChange={onChange}
+        />
+      )}
```

Comment:
This is a really bad way of doing it. What I would recommend is doing what you did for the other attributes such as id, name, value and onChange. Just create a "type" props for the CustomTextInput component, and pass it when directly to the <input> element you have here. Its going to make your code much simpler.

When you're making a custom <input> component like in this case, you essentially want to make a wrapper around a <input>, and let the client code give the values it wants to set to the wrapped <input> element.

Reply:
Sounds good, we were debating between either doing the prop or doing it using spread operator as above since we didn't think we needed anything more than text and password, but we can make the change you've suggested

Reply:
Even if we won't need a type other than text or password, not having an if else if going to make the code easier to maintain

Reply:
Yeah, feel free to change up the custom components as requirements evolve. I would suggest doing this since it would probably be text/password only:
```suggestion
        <input
          className={`custom-text-input input-border ${hasErrors ? 'error' : ''}`}
          id={fieldName}
          name={fieldName}
          data-testid="custom-text-input"
          type={password ? "password": "text"}
          value={value}
          onChange={onChange}
        />

```
`password` would be a prop and you would use it like this:

Password:
```jsx
        <CustomTextInput
          title="Password"
          fieldName="password"
          value={value}
          onChange={handleChange}
          hasErrors={checkFunction}
          errorMessage={errorMessage}
          password
        />
```
Text:
```jsx
        <CustomTextInput
          title="Text"
          fieldName="text"
          value={value}
          onChange={handleChange}
          hasErrors={checkFunction}
          errorMessage={errorMessage}
        />
```

Reply:
They could do that. But I think that having them pass the type as a prop would make the custom component bahv more closely to a real `<input>` element, which would make it easier to use.